src/main/java/CodeBuildException.java

```diff
@@ -0,0 +1,14 @@
+import org.jenkinsci.plugins.scriptsecurity.sandbox.whitelists.Whitelisted;
+
+public class CodeBuildException extends InterruptedException{
+    private final CodeBuildResult codeBuildResult;
+
+    public CodeBuildException(CodeBuildResult codeBuildResult) {
+        this.codeBuildResult = codeBuildResult;
```

Comment:
We wrap the `CodeBuildResult` in an exception, because we may need to access artifacts even when there is an exception.